### PR TITLE
[Golang][client] Make it possible to generate go.mod and go.sum files

### LINKED ISSUE #2796: [REQ][Golang][client] Add go.mod to go client templates
### Is your feature request related to a problem? Please describe.

I need to use the autogenerated go client as a dependency of another go module, which is much easier when it's a module as well.

## Describe the solution you'd like

I'd like the go client templates to also contain go.mod/go.sum files that would be rendered (not sure if this should be by default, but it probably wouldn't hurt). To do that, the go config file would also need to contain the full module name, e.g. "github.com/user/mymodule".

## Describe alternatives you've considered

I could maintain these files by hand outside of openapi-generator functionality.

## Additional context

This seems to be a simple enough modification that I would probably be able to send a PR myself, but first I'd like to hear if it's wanted/acceptable.

### PR DESCRIPTION
**Pull request by bkabrda**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@antihax @bvwells @grokify @kemokemo

### Description of the PR

Fixes #2796.
I didn't regenerate samples, since this is only triggered when moduleName is specified (either in config or on commandline). If you wish, I can definitely add this option to the scripts that generate samples and amend this PR.

I do have one question - it seems that the files under docs/generators are autogenerated, but I failed to find a script that generates them. Am I just looking wrong or are these handwritten? Thanks!

### CODE REVIEW

**Comment by etherealjoy** [comment]
I think we need this also for go server.

**Comment by bkabrda** [comment]
So after a Gitter discussion with @wing328 I made this use GIT_USER_ID and GIT_REPO_ID and generate go.mod and go.sum by default.

File: modules/openapi-generator/src/main/resources/go/go.mod.mustache [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module github.com/{{gitUserId}}/{{gitRepoId}}
```
Comment (bvwells) [inline modules/openapi-generator/src/main/resources/go/go.mod.mustache]: This all seems to be very specific to modules in GitHub repos. Personally I would think about making the complete module path as a input variable to the generator...

**Comment by bvwells** [comment]
Hi @bkabrda, missed the thread on gitter, but this change makes the module definition very specific to GitHub repos. Might want to rethink the use of GIT_USER_ID and GIT_REPO_ID as the input to the generator. Perhaps specify the module path as a single path to the generator...

File: modules/openapi-generator/src/main/resources/go/go.mod.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/go/go.mod.mustache]: Yup, we talked about this as well. No one has asked for supporting other git hosting services so far (which is a bit surprising to me). We can introduce another option "gitHost" if there's a need to support Gitlab, BitBucket or other Git hosting services.

Comment (bvwells) [reply modules/openapi-generator/src/main/resources/go/go.mod.mustache]: Okay, fair enough. For me, it feels a bit clunky to have to specify these two (and perhaps in future three)  pieces of information separately on the command line.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/go/go.mod.mustache]: Yup, agreed some users prefer just one single option for this particular setting. gitUserID and gitRepoID are global options used by other generators as well (e.g. PHP) so another benefit is consistency among generators. Let's go with the current approach and see if more Go users have an opinion on this.

**Pull request merged**